core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -379,14 +379,20 @@ private void updateMembers() {
       }
    }
 
-   public synchronized void doHandleClusterView() {
+   public synchronized void doHandleClusterView(int viewId) {
       // TODO Clean up ClusterCacheStatus instances once they no longer have any members
       if (currentTopology == null)
          return;
 
       List<Address> newClusterMembers = transport.getMembers();
+      int newViewId = transport.getViewId();
+      if (newViewId != viewId) {
```

Comment:
Is it possible to get a viewId from coordinator before our transport has updated its own?

Reply:
Yes, the topology commands are OOB so they could be delivered before the view message. But this is not about receiving the view later, it's about receiving a newer view while confirming all the members are available. I'll add a comment.